### PR TITLE
feat(deploy): allow updating an existing deploy

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Related to https://github.com/netlify/cli/issues/1236

Allows passing in an existing deploy to update.
This is required since deploying edge handlers must be done before the deploy is complete.
At the moment the whole deploy cycle happens in the `js-client` which doesn't allow deploying edge handlers.
This change allows creating the deploy in the CLI and finalising it in the `js-client`.

**- Test plan**

Ran different variations of the deploy command from the CLI.

**- Description for the changelog**

Allow updating an existing deploy when invoking the deploy function.

**- A picture of a cute animal (not mandatory but encouraged)**

🦁

### CODE REVIEW

File: src/deploy/index.js [hunk 1]
```diff
@@ -75,8 +77,14 @@ module.exports = async (api, siteId, dir, opts) => {
       draft: opts.draft
     }
   })
+  if (opts.deployId === null) {
+    deployParams = { ...deployParams, ...(title && { title }) }
```
Comment (mraerino) [inline src/deploy/index.js:81]:
```suggestion
    if(title) {
      deployParams = { ...deployParams, title }
    }
```

File: src/deploy/index.js [hunk 2]
```diff
@@ -75,8 +77,16 @@ module.exports = async (api, siteId, dir, opts) => {
       draft: opts.draft
     }
   })
+  if (opts.deployId === null) {
+    if(title) {
```
Comment (ehmicky) [inline src/deploy/index.js:81]: Quick question: wouldn't be simpler to keep `title` in `cleanDeep()`?

Comment (erezrokah) [reply src/deploy/index.js:81]: Do you mean under the `opts.deployId === null` condition? 
https://open-api.netlify.com/#operation/updateSiteDeploy doesn't accept `title` (unless I'm missing something), so didn't want to do it outside the condition.
We can do it inside the condition, but didn't see a reason to use the dependency for just one property.
I could go either way.

Comment (ehmicky) [reply src/deploy/index.js:81]: Oh, makes sense! :+1:

**Review by ehmicky — APPROVED** [review]

**Pull request merged**